src/hooks/useDynamicOptionsLoaderState.tsx

```diff
@@ -166,25 +169,37 @@ const LoadOptionsSet = React.memo(function LoadOptionsSet({
   form,
   optionsSetResult,
   setOptionsSetResults,
+  formIsReadOnly,
 }: {
   form: FormTypes.Form
   optionsSetResult: OptionsSetResult
   setOptionsSetResults: React.Dispatch<React.SetStateAction<OptionsSetResult[]>>
+  formIsReadOnly: boolean
 }) {
   const hasOptionsSet = React.useMemo<boolean>(() => {
     return !!formElementsService.findFormElement(
       form.elements,
       (formElement) => {
         const formElementWithOptions =
           typeCastService.formElements.toOptionsElement(formElement)
+
+        // for read only forms we don't want to fetch if we don't have to
+        if (
+          formIsReadOnly &&
```

Comment:
I dont actually think we need this... 🤔

Reply:
mmm you might be right. Will a clarification use the previous definition or the current form def?

Reply:
clarification will use current form def

Reply:
removing...